src/main/java/uk/gov/hmcts/ccd/domain/types/TextCaseReferenceCaseLinkValidator.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+import uk.gov.hmcts.ccd.domain.service.common.CaseService;
+import uk.gov.hmcts.ccd.endpoint.exceptions.ResourceNotFoundException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.Collections;
+import java.util.List;
+
+@Named("TextCaseReferenceCaseLinkValidator")
+@Singleton
+public class TextCaseReferenceCaseLinkValidator implements PredefinedTypeFieldValidator {
+
+    private CaseService caseService;
+    private TextValidator textValidator;
+
+    @Inject
+    public TextCaseReferenceCaseLinkValidator(@Qualifier("TextValidator") TextValidator textValidator, CaseService caseService) {
+        this.caseService = caseService;
+        this.textValidator = textValidator;
+    }
+
+    @Override
+    public List<ValidationResult> validate(final String dataFieldId,
+                                           final JsonNode dataValue,
+                                           final CaseFieldDefinition caseFieldDefinition) {
+
+        List<ValidationResult> validationResults = textValidator.validate(dataFieldId, dataValue, caseFieldDefinition);
+        if (validationResults.isEmpty() && !textValidator.isNullOrEmpty(dataValue)) {
+            final String value = dataValue.textValue();
+            return isAnExistingCase(value, dataFieldId);
+        }
+        return validationResults;
+    }
+
+    private List<ValidationResult> isAnExistingCase(final String value, final String dataFieldId) {
+        try {
+            this.caseService.getCaseDetailsByCaseReference(formatCaseReference(value));
+            return Collections.emptyList();
+        } catch (ResourceNotFoundException resourceNotFoundException) {
```

Comment:
what happens in case of other exceptions that are not ResourceNotFoundException? we should test that out and decide what happens in that case

Reply:
There are some exception in uk.gov.hmcts.ccd.endpoint.exceptions .  Should we  throw BadRequestException() for unexpected errors ?

Reply:
DONE

Reply:
DONE